arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,360 @@
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        Section.allCases[section].titleForHeader
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        Section.allCases.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch Section.allCases[section] {
+        case .included:
+            return includedPhases.count
+        case .excluded:
+            return excludedPhases.count
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: UITableViewCell
+        let phase: AGSCodedValue
+        switch Section.allCases[indexPath.section] {
+        case .included:
+            cell = tableView.dequeueReusableCell(withIdentifier: Section.included.cellIdentifier, for: indexPath)
+            phase = includedPhases[indexPath.row]
+            cell.textLabel?.text = "Phase: " + phase.name
+            if let summary = summaries[phase] {
+                cell.detailTextLabel?.text = "C: " + numberFormatter.string(from: NSNumber(value: summary.totalCustomers))! + "    L: " + numberFormatter.string(from: NSNumber(value: summary.totalLoad))!
+            } else {
+                cell.detailTextLabel?.text = "N/A"
+            }
+        case .excluded:
+            cell = tableView.dequeueReusableCell(withIdentifier: Section.excluded.cellIdentifier, for: indexPath)
+            phase = excludedPhases[indexPath.row]
+            cell.textLabel?.text = phase.name
+        }
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCell.EditingStyle {
+        switch Section.allCases[indexPath.section] {
+        case .included:
+            return .delete
+        case .excluded:
+            return .insert
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        tableView.beginUpdates()
+        switch editingStyle {
+        case .delete:
+            tableView.deleteRows(at: [indexPath], with: .automatic)
+            let phase = includedPhases[indexPath.row]
+            includedPhases.remove(at: indexPath.row)
+            // Binary search to find the insertion index.
+            let insertionIndex = (excludedPhases as NSArray).index(
+                of: phase,
+                inSortedRange: NSRange(excludedPhases.indices),
+                options: [.insertionIndex],
+                usingComparator: { ($0 as! AGSCodedValue).name.compare(($1 as! AGSCodedValue).name) }
+            )
+            excludedPhases.insert(phase, at: insertionIndex)
+            let section = Section.excluded.rawValue
+            tableView.insertRows(at: [IndexPath(row: insertionIndex, section: section)], with: .automatic)
```

Comment:
You could help yourself out here (and elsewhere) with an extension of `IndexPath`:

```swift
private extension IndexPath {
    init(row: Int, section: CreateLoadReportViewController.Section) {
        self.init(row: row, section: section.rawValue)
    }
}

tableView.insertRows(at: [IndexPath(row: insertionIndex, section: .excluded)], with: .automatic)
```